### PR TITLE
feat(graph): add Kuzu bootstrap wiring

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- add typed graph-store config and env validation for a Kuzu-backed graph projection store
- wire a Kuzu graph-store dependency into bootstrap opening/closing so startup can fail fast on bad graph-store configuration
- add a minimal embedded Kuzu adapter plus unit coverage for graph-store dependency wiring and ping behavior

## Local testing
- `make verify`
- started `./bin/cerebro serve` locally with `CEREBRO_GRAPH_STORE_DRIVER=kuzu` and `CEREBRO_KUZU_PATH=$(mktemp -d)/graph`
- hit `/health` and verified `graph_store` reported `ready`

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review this PR after the latest fixes. Focus on whether the prior code review and security findings are resolved, and call out any remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed merge-conflict resolution and config test isolation fix. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

File: internal/graphstore/kuzu/kuzu.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		return nil, errors.New("kuzu path is required")
+	}
+	absPath, err := filepath.Abs(rawPath)
+	if err != nil {
+		return nil, fmt.Errorf("resolve kuzu path: %w", err)
+	}
+	if err := os.MkdirAll(filepath.Dir(absPath), 0o755); err != nil {
+		return nil, fmt.Errorf("create kuzu parent directory: %w", err)
+	}
+	dsn := "kuzu://" + filepath.ToSlash(absPath)
```
Comment (josh-writer) [inline internal/graphstore/kuzu/kuzu.go]: Escape the filesystem path before building the Kuzu DSN. go-kuzu parses the DSN with net/url.Parse and opens u.Path, so valid paths containing ?, #, or malformed % escapes can fail, truncate, or open the wrong database path.

**Review by josh-writer — CHANGES_REQUESTED** [review]

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Re-audited the live unresolved thread (including outdated) against the current branch tip. The Kuzu DSN path escaping concern is already addressed by `kuzuDSN()` constructing a `url.URL` path plus the regression coverage in `TestKuzuDSNEscapesFilesystemPath`, so no code change was needed. Validation run: none locally during this audit; existing branch `verify` is green. Remaining unresolved threads after this audit: 0.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review Kuzu bootstrap wiring deeply, with special attention to integration boundaries:

1. **Configuration and startup safety**: verify driver selection, missing Kuzu paths, relative/absolute paths, permissions, and fallback behavior do not silently boot with the wrong graph backend.
2. **Resource lifecycle**: check Open/Ping/Close ordering, repeated bootstrap calls, test cleanup, connection leaks, and whether failures degrade health correctly.
3. **CGO/build tags**: make sure cgo and non-cgo builds both compile and fail predictably, including CI/Linux/macOS differences and vendored dependency assumptions.
4. **Schema and concurrency**: review schema initialization races, idempotent table creation, simultaneous bootstrap/test access, and migration compatibility with later projection PRs.
5. **Security boundaries**: ensure path config cannot be abused for unexpected filesystem writes and that graph errors do not leak sensitive environment/config values.
6. **Stack compatibility**: verify the wiring remains compatible with downstream source projection, local graph traversal tests, and runtime sync PRs.
